### ISSUE TITLE
Pre-Training Tasks for Embedding-Based Large-Scale Retrieval

### ISSUE THREAD

**Issue by pocca2048** [pocca2048/ML-paper-reading#9]
**Conference** : 2020 ICLR
**Link** : https://arxiv.org/pdf/2002.03932.pdf
**Authors' Affiliation** : cmu, google
**TL;DR** : 적절히 디자인된 paragraph-level 사전학습 태스크로 학습하면, transformer 모델은 bm25보다 훨씬 잘 할 수 있다.

**Summary** :
> With adequately designed paragraph-level pre-training tasks, the Transformer models can remarkably improve over the widely-used BM-25 as well as embedding models without Transformers.

* Inverse Cloze Task (ICT)
* Body First Selection (BFS)
* Wiki Link Prediction (WLP)

# 1 Introduction

retrieval phase + scoring phase

retrieval은 중요하다. 높은 recall을 가져야 한다. 그러면서 매우 efficient 하지 않으면 쓰지도 못한다.

token-based matching 알고리즘 방식과 embedding based 방식이 있음.

embedding based 방식을 two-tower retrieval model, Siamese network, dual-encoder model 로도 부름.

The token-level masked-LM (MLM) pretraining task is crucial to the success of BERT-style cross-attention models. Nevertheless, what pre-training tasks are useful for improving two-tower Transformer models in large-scale retrieval, remains a crucial yet unsolved research problem. In this paper, we aim to answer this question by studying different pre-training tasks for the two-tower Transformer models.

Contribution:

* Paragraph-level pre-training tasks such as Inverse Cloze Task (ICT), Body First Selection (BFS), and Wiki Link Prediction (WLP) hugely improve the retrieval quality, whereas the most widely used pre-training task (the token-level masked-LM) gives only marginal gains.
* 적절한 pretraining하면 bi encoder가 bm25를 크게 이김.
* BoW MLP보다 transformer가  paragraph-level pre-training으로부터 더 큰 이득을 얻음.

# 3 Pretraining Tasks of Different Semantic Granularities

pre-training data is defined as positive query-document (q, d) pair

A good pre-training task : 1) it should be relevant to the downstream task 2) It should be cost-efficient to collect the pre-training data, ideally not requiring additional human supervision.

제안:

1. Inverse Cloze Task (ICT), 
2. Body First Selection (BFS)
3. Wiki Link Prediction (WLP). 

BFS and WLP를 새로 제안한것. The training data for all these tasks can be freely obtained based from Wikipedia without an additional manual labeling process.

Inverse Cloze Task (ICT) Given a passage p consisting of n sentences, p = {s1, . . . , sn}, the query q is a sentence randomly drawn from the passage, q = si , i ∼ [1, n], and the document d is the rest of sentences, d = {s1, . . . , si−1, si+1, . . . , sn}. See (q1,d) in Figure 2 as an example. This task captures the semantic context of a sentence and was originally proposed by Lee et al. (2019). 

Body First Selection (BFS) We propose BFS to capture semantic relationship outside of the local paragraph. Here, the query q2 is a random sentence in the first section of a Wikipedia page, and the document d is a random passage from the same page (Figure 2). Since the first section of a Wikipedia article is often the description or summary of the whole page, we expect it to contain information central to the topic. 

Wiki Link Prediction (WLP) We propose WLP to capture inter-page semantic relation. The query q3 is a random sentence in the first section of a Wikipedia page, and the document d is a passage from another page where there is a hyperlink link to the page of q3 (Figure 2). Intuitively, a hyperlink link indicates relationship between the two Wikipedia pages. Again, we take a sentence from the first section because it is often the description or summary of the topic.

# 4 Experiments

신기한게 MLM보다 여기서 나온 pretraining process의 성능이 월등히 좋음. 다만, pretraining task별로 성능이 안나온건 매우 사기. 그리고 pretrain한 BoW-MLP의 성능이 너무 좋은데? 데이터가 적을때는 MLM 학습한 transformer보다 좋다.

(Specifically, BoW-MLP looks up uni-grams from the embedding table5 , aggregates the embeddings with average pooling, and passes them through a shallow two-layer MLP network with tanh activation to generate the final 512-dimensional query/document embeddings. For fair comparison, the BoW-MLP encoder has a comparable model size to the Transformer encoder (i.e., 128M v.s. 110M parameters, slightly favorable to BoW-MLP encoder).)

실험결과를 보면 역시나 ICT가 제일 좋고 나머지는 하나마나 거의 의미없음. 그럼 이 논문의 컨트리뷰션은 사실상 그냥 실험만 한것.

# 5 CONCLUSION

random init , MLM 같은 부적절한 사전학습 < BM25 베이스라인

적절히 디자인된 사전학습이 중요하다.
